salt/modules/saltutil.py

```diff
@@ -1273,7 +1273,7 @@ def _exec(client, tgt, fun, arg, timeout, tgt_type, ret, kwarg, **kwargs):
         _cmd = client.cmd_subset
         cmd_kwargs = {
             'tgt': tgt, 'fun': fun, 'arg': arg, 'tgt_type': tgt_type,
-            'ret': ret, 'kwarg': kwarg, 'sub': kwargs['subset'],
+            'ret': ret, 'cli': True, 'kwarg': kwarg, 'sub': kwargs['subset'],
```

Comment:
I don't understand why this is the correct fix. Could you explain in more detail?

Reply:
This is how subset was originally implemented for the `salt` command line

https://github.com/saltstack/salt/commit/27cc8295e69

So calling it from the saltutil command options need to call it the same way.

https://github.com/saltstack/salt/blob/2017.7/salt/cli/salt.py#L144

Reply:
Can we either refactor this then so there is a single source of truth for this behavior or heavily document in both places why this behavior is necessary? When we have these mystery options being passed with no clear documentation on why they are necessary or what they do, it feels a little tech debt-y to me. :]

Reply:
:+1: